sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -679,7 +679,25 @@ class SQLContext(@transient val sparkContext: SparkContext)
    * @since 1.3.0
    */
   def dropTempTable(tableName: String): Unit = {
-    cacheManager.tryUncacheQuery(table(tableName))
+    dropTempTable(tableName, ifExists = false)
+  }
+
+  /**
+   * Drops the temporary table with the given table name in the catalog. If the table has been
+   * cached/persisted before, it's also unpersisted.
+   *
+   * @param tableName the name of the table to be unregistered.
+   * @param ifExists whether ignore non-existent table
+   *
+   * @group basic
+   * @since 1.5.0
+   */
+  def dropTempTable(tableName: String, ifExists: Boolean): Unit = {
```

Comment:
If using Try, we have to look into the return to check the reason. Not only non-existent table leads to failure of drop operation. This would only ignore the certain case.

I'll close it if not necessary.